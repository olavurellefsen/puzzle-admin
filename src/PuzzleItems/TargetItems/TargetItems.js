import React from "react";
import { TargetItemsContainer, TargetItemsImage } from "./TargetItems.style";

export default () => (
  <TargetItemsContainer data-testid="TargetItemsContainer">
    {" "}
    <TargetItemsImage
      src={"https://via.placeholder.com/200x800?text=Target"}
      data-testid="TargetItemsImage"
    />
  </TargetItemsContainer>
);
