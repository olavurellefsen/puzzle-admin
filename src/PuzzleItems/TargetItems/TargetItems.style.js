import styled from 'styled-components';

export const TargetItemsContainer = styled.div`
  margin: 0px 10px;
`;

export const TargetItemsImage = styled.img`
`;