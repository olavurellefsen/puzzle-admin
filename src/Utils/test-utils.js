import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
