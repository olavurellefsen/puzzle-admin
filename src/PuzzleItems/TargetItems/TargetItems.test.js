import React from "react";
import { render, cleanup } from 'test-utils';
import TargetItems from "./TargetItems";
afterEach(cleanup)

it("should render correctly", () => {
  const { container } = render(<TargetItems />);
  expect(container).toMatchSnapshot();
});

it("should show the TargetItemsContainer element", () => {
  const { queryByTestId } = render(<TargetItems />);
  expect(queryByTestId("TargetItemsContainer")).toBeTruthy();
});

it("should show the TargetItemsImage element", () => {
  const { queryByTestId } = render(<TargetItems />);
  expect(queryByTestId("TargetItemsImage")).toBeTruthy();
});