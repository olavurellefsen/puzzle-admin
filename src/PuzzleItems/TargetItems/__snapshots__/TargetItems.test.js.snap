// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa dUCXaL"
    data-testid="TargetItemsContainer"
  >
     
    <img
      class="sc-bwzfXH bFHdpU"
      data-testid="TargetItemsImage"
      src="https://via.placeholder.com/200x800?text=Target"
    />
  </div>
</div>
`;
